zerver/lib/actions.py

```diff
@@ -2010,6 +2011,20 @@ def do_change_left_side_userlist(user_profile, setting_value, log=True):
         log_event(event)
     send_event(event, [user_profile.id])
 
+def mark_user_star_backend(user_profile, email):
+    starred_user = get_object_or_404(UserProfile, email=email)
+    result = {}
+    if not starred_user in user_profile.starred_users.all():
```

Comment:
Instead of toggling, the backend should take what state you're trying to set it to. If your internet connection is slow, you might click "star user" twice, which would toggle twice and ultimately unstar the user.